src/lib/createStardustComponent.tsx

```diff
@@ -0,0 +1,29 @@
+import createComponentInternal from './createComponent'
+import * as React from 'react'
+import { pick } from 'lodash'
+import { ComponentSlotClasses } from '@stardust-ui/react'
+
+export interface RenderStardustResultConfig {
+  classes: ComponentSlotClasses
+  rtl: boolean
+}
+
+export interface CreateStardustComponentConfig<P> {
+  displayName?: string
+  render: (config: RenderStardustResultConfig, props: P) => React.ReactNode
```

Comment:
As a I remember, we decided  to go with one argument:

```diff
- render: (config: RenderStardustResultConfig, props: P) => React.ReactNode
+ render: (props: P & { stardust: RenderStardustResultConfig }) => React.ReactNode
```

Reply:
I left this comment regarding that: https://github.com/stardust-ui/react/pull/503#discussion_r235078288 but I am open to change the implementation if there is a reason why we want that.

Reply:
We have a discussion about this with Levi. The signature with one argument matches an `React.SFC` signature:

```jsx
export const A: React.SFC = (props) => <div />

export const B = createComponent({
  render: A,
})
```

A little crazy, but it will work. Note sure that this will not break rendering, but SFC is a same render function.

Reply:
Got it, okay, I will change it. Thanks for the clarification.

Reply:
Changed!